File: Angular_Spring/Challenge/Resolved/back/src/main/java/deep/session/CA_CIB/controller/MovieController.java
package deep.session.CA_CIB.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;
import deep.session.CA_CIB.model.Movie;
import deep.session.CA_CIB.service.MovieService;

import jakarta.validation.Valid;
import java.util.List;

@RestController
@RequestMapping("/movies")
@Validated
public class MovieController {
    @Autowired
    private MovieService movieService;

    // Récupérer tous les films
    @GetMapping
    public List<Movie> getAllMovies() {
        return movieService.getAllMovies();
    }

    // Récupérer un film par ID
    @GetMapping("/{id}")
    public ResponseEntity<Movie> getMovieById(@PathVariable Long id) {
        Movie movie = movieService.getMovieById(id);
        return new ResponseEntity<>(movie, HttpStatus.OK);
    }

    // Créer un nouveau film
    @PostMapping
    public ResponseEntity<String> createMovie(@Valid @RequestBody Movie movie) {
        movieService.saveMovie(movie);
        return new ResponseEntity<>("Movie created successfully", HttpStatus.CREATED);
    }

    // Mettre à jour un film existant
    @PutMapping("/{id}")
    public ResponseEntity<String> updateMovie(@PathVariable Long id, @Valid @RequestBody Movie movie) {
        movie.setId(id);
        movieService.saveMovie(movie);
        return new ResponseEntity<>("Movie updated successfully", HttpStatus.OK);
    }

    // Supprimer un film par ID
    @DeleteMapping("/{id}")
    public ResponseEntity<String> deleteMovie(@PathVariable Long id) {
        movieService.deleteMovie(id);
        return new ResponseEntity<>("Movie deleted successfully", HttpStatus.OK);
    }
}

File: Python_Angular_Typescript/02 - Challenge/Resolved/back/src/controller/movie_controller.py
from typing import List
from fastapi import APIRouter, HTTPException
from src.model.movie import Movie
from src.service.movie_service import MovieService

router = APIRouter()

movie_service = MovieService()

@router.get("/movies", response_model=List[Movie])
def get_all_movies():
    return movie_service.get_all_movies()

@router.get("/movies/{movie_id}", response_model=Movie)
def get_movie_by_id(movie_id: int):
    return movie_service.get_movie_by_id(movie_id)

@router.post("/movies", response_model=Movie)
def create_movie(movie: Movie):
    return movie_service.save_movie(movie)

@router.patch("/movies/{movie_id}", response_model=Movie)
def update_movie(movie_id: int, movie: Movie):
    return movie_service.update_movie(movie_id, movie)

@router.delete("/movies/{movie_id}")
def delete_movie(movie_id: int):
    movie_service.delete_movie(movie_id)
    return {"message": "Movie deleted successfully"}